libs/utils/analysis/latency_analysis.py

```diff
@@ -286,6 +286,42 @@ def cr(row):
         run_df = run_df[run_df.next_state.isin(['S', 'x'])][['running_time']]
         return run_df
 
+    @memoized
+    def _dfg_task_residency(self, task):
+        """
+        DataFrame of a task's execution time on each CPU
+
+        The returned DataFrame index is the CPU indexes
+        The DataFrame has just one column:
+        - runtime: the time the task spent being active (running or waking up)
+          on a given CPU, in seconds.
+
+        :param task: the task to report runtimes for
+        :type task: int or str
+        """
+        cpus = range(self._platform['cpus_count'])
+        runtimes = {cpu : 0.0 for cpu in cpus}
+
+        df = self._dfg_latency_df(task)
+
+        # Exclude sleep time
+        df = df[df.curr_state != 'S']
+
+        for time, data in df.iterrows():
+            cpu = data['__cpu']
+            if np.isnan(cpu):
+                if data['curr_state'] == 'W':
+                    cpu = data['target_cpu']
+                else:
+                    self._log.warning(
```

Comment:
Kind of worrying: what means "results will be affected"?!?

Reply:
I think I mostly mean "couldn't get any relevant data for a timestamp". I've never seen that happen so maybe this should be an exception rather than a warning ?

Reply:
yes... if we get not usable data... better to raise.